build-system/babel-plugins/babel-plugin-jsx-style-object/index.js

```diff
@@ -0,0 +1,181 @@
+/**
+ * @fileoverview
+ * Converts Object Expression syntax in `style={{foo: 'bar'}}` to a concatenated
+ * string expression.
+ *
+ * This only transforms a file if it imports the `baseModule` for JSX below,
+ * which lacks runtime support for converting the Object into a string.
+ * This transform takes that responsibility instead.
+ */
+
+const baseModule = 'core/dom/jsx';
+const helperModule = 'core/dom/jsx-style-property-string';
+const helperFn = 'jsxStylePropertyString';
+
+const nonDimensional = new Set([
+  'animation-iteration-count',
+  'box-flex',
+  'box-flex-group',
+  'box-ordinal-group',
+  'column-count',
+  'fill-opacity',
+  'flex',
+  'flex-grow',
+  'flex-positive',
+  'flex-shrink',
+  'flex-negative',
+  'flex-order',
+  'font-weight',
+  'line-clamp',
+  'line-height',
+  'opacity',
+  'order',
+  'orphans',
+  'stop-opacity',
+  'stroke-dashoffset',
+  'stroke-opacity',
+  'stroke-width',
+  'tab-size',
+  'widows',
+  'z-index',
+  'zoom',
+]);
+
+module.exports = function (babel) {
+  const {types: t} = babel;
+
+  const dashCase = (camelCase) =>
+    camelCase.replace(/[A-Z]/g, '-$&').toLowerCase();
+
+  let hasBaseModule = false;
+  let hasHelperModule = false;
+  let programPath = null;
+
+  /**
+   * @param {babel.NodePath<import('@babel/types').ObjectProperty>} path
+   * @return {?babel.Node}
+   */
+  function transformProp(path) {
+    // @ts-ignore
+    const name = path.node.key.name || path.node.key.value;
+    const cssName = dashCase(name);
```

Comment:
Likely need to support vendor-prefixed fields? `{ webkitFoo }` should map to `-webkit-foo`

Reply:
This is already handled the same way as Preact, by requiring capitalized `WebkitFoo` so that `W` is converted to `-w`.

Reply:
I think that's going to be surprising for most people? We'd need a linter to warn about it.

Reply:
It's standard:

> https://reactjs.org/docs/dom-elements.html#style
> Vendor prefixes other than `ms` should begin with a capital letter. 

If we'd like to lint it it's probably a separate concern than this change.